x-pack/plugins/profiling/public/components/stacked_bar_chart.tsx

```diff
@@ -5,47 +5,109 @@
  * 2.0.
  */
 
+import {
+  Axis,
+  BrushAxis,
+  Chart,
+  HistogramBarSeries,
+  Settings,
+  StackMode,
+  timeFormatter,
+  TooltipInfo,
+  XYChartElementEvent,
+} from '@elastic/charts';
+import { EuiPanel } from '@elastic/eui';
 import React, { useContext } from 'react';
-
-import { Axis, BarSeries, Chart, Settings } from '@elastic/charts';
-
-import { timeFormatter } from '@elastic/charts';
-
+import { TopNSample, TopNSubchart } from '../../common/topn';
+import { chartTheme } from '../utils/chart_styles';
+import { asPercentage } from '../utils/formatters/as_percentage';
 import { TopNContext } from './contexts/topn';
+import { SubChart } from './subchart';
+
+function SubchartTooltip({
+  highlightedSubchart,
+}: TooltipInfo & { highlightedSubchart: TopNSubchart }) {
+  return (
+    <EuiPanel>
+      <SubChart
+        index={highlightedSubchart.Index}
+        color={highlightedSubchart.Color}
+        category={highlightedSubchart.Category}
+        percentage={highlightedSubchart.Percentage}
+        data={highlightedSubchart.Series}
+        height={200}
+        width={400}
+      />
+    </EuiPanel>
+  );
+}
 
 export interface StackedBarChartProps {
-  id: string;
-  name: string;
   height: number;
-  x: string;
-  y: string;
-  category: string;
+  asPercentages: boolean;
+  onBrushEnd: (range: { rangeFrom: string; rangeTo: string }) => void;
+  onSampleClick: (sample: TopNSample) => void;
+  onSampleOut: () => void;
+  highlightedSubchart?: TopNSubchart;
 }
 
 export const StackedBarChart: React.FC<StackedBarChartProps> = ({
-  id,
-  name,
   height,
-  x,
-  y,
-  category,
+  asPercentages,
+  onBrushEnd,
+  onSampleClick,
+  onSampleOut,
+  highlightedSubchart,
 }) => {
   const ctx = useContext(TopNContext);
 
   return (
     <Chart size={{ height }}>
-      <Settings showLegend={false} />
-      <BarSeries
-        id={id}
-        name={name}
-        data={ctx.samples}
-        xAccessor={x}
-        yAccessors={[y]}
-        stackAccessors={[x]}
-        splitSeriesAccessors={[category]}
+      <Settings
+        showLegend={false}
+        tooltip={{
+          customTooltip: highlightedSubchart
+            ? (props) => <SubchartTooltip {...props} highlightedSubchart={highlightedSubchart} />
+            : () => <></>,
+        }}
+        brushAxis={BrushAxis.X}
+        onBrushEnd={(brushEvent) => {
+          const rangeFrom = new Date(brushEvent.x![0]).toISOString();
+          const rangeTo = new Date(brushEvent.x![1]).toISOString();
+
+          onBrushEnd({ rangeFrom, rangeTo });
+        }}
+        theme={chartTheme}
+        onElementClick={(events) => {
+          const [value] = events[0] as XYChartElementEvent;
+          onSampleClick(value.datum as TopNSample);
+        }}
+        onElementOver={() => {
+          onSampleOut();
+        }}
+        onElementOut={() => {
+          onSampleOut();
+        }}
       />
+      {ctx.charts.map((chart) => (
+        <HistogramBarSeries
```

Comment:
I think we should also align the time axis to the new multi-layer time axis used in discover.
If you want I can provide you with the right settings for that.
Please consider that the default time axis can leave some white space to the right of the chart to accommodate for all the labels, where the new multi-layer time axis don't overflow the chart area

Reply:
sounds good to me, can you help me find those settings?

Reply:
I can push a commit for that later today